Branch-SDK/src/io/branch/referral/BranchShortLinkBuilder.java

```diff
@@ -0,0 +1,152 @@
+    /**
+     * <p> Sets the alias for this link. </p>
+     *
+     * @param alias Link 'alias' can be used to label the endpoint on the link.
+     *              <p>
+     *              For example:
+     *              http://bnc.lt/AUSTIN28.
+     *              Should not exceed 128 characters
+     *              </p>
+     * @return A {@link BranchShortLinkBuilder} instance.
+     */
+    public BranchShortLinkBuilder setAlias(String alias) {
+        this.alias_ = alias;
+        return this;
+    }
+
+    /**
+     * <p> Sets the channel for this link. </p>
+     *
+     * @param channel A {@link String} denoting the channel that the link belongs to. Should not
+     *                exceed 128 characters.
+     * @return A {@link BranchShortLinkBuilder} instance.
+     */
+    public BranchShortLinkBuilder setChannel(String channel) {
+        this.channel_ = channel;
+        return this;
+    }
+
+    /**
+     * <p> Sets the amount of time that Branch allows a click to remain outstanding.</p>
+     *
+     * @param duration A {@link Integer} value specifying the time that Branch allows a click to
+     *                 remain outstanding and be eligible to be matched with a new app session.
+     * @return A {@link BranchShortLinkBuilder} instance.
+     */
+    public BranchShortLinkBuilder setDuration(int duration) {
+        this.duration_ = duration;
+        return this;
+    }
+
+    /**
+     * <p> Set a name that identifies the feature that the link makes use of.</p>
+     *
+     * @param feature A {@link String} value identifying the feature that the link makes use of.
+     *                Should not exceed 128 characters.
+     * @return A {@link BranchShortLinkBuilder} instance.
+     */
+    public BranchShortLinkBuilder setFeature(String feature) {
+        this.feature_ = feature;
+        return this;
+    }
+
+    /**
+     * <p> Set the parameters associated with the link.</p>
+     *
+     * @param parameters A {@link JSONObject} value containing the deep linked params associated with
+     *                   the link that will be passed into a new app session when clicked.
+     *                   {@see addParameters} if you want to set parameters as individual key value.
+     * @return A {@link BranchShortLinkBuilder} instance.
+     */
+    public BranchShortLinkBuilder setParameters(JSONObject parameters) {
+        this.params_ = parameters;
+        return this;
+    }
+
+    /**
+     * <p>Set a name that identify the stage in an application or user flow process.</p>
+     *
+     * @param stage A {@link String} value identifying the stage in an application or user flow
+     *              process. Should not exceed 128 characters.
+     * @return A {@link BranchShortLinkBuilder} instance.
+     */
+    public BranchShortLinkBuilder setStage(String stage) {
+        this.stage_ = stage;
+        return this;
+    }
+
+    /**
+     * <p>Set the number of times the link should perform deep link.</p>
```

Comment:
This is a confusing description. The user passes an `int` into this function, so saying "the number of times it deep links" (which is wrong, it's how many times reward rules will be triggered) indicates that I can set that to an arbitrary number and should see that as the result.